zerver/tests/test_submessage.py

```diff
@@ -200,3 +200,22 @@ def test_submessage_event_sent_after_transaction_commits(self) -> None:
                     "Events should be sent only after the transaction commits."
                 )
                 do_add_submessage(hamlet.realm, hamlet.id, message_id, "whatever", "whatever")
+
+    def test_fetch_message_containing_submessages(self) -> None:
+        cordelia = self.example_user("cordelia")
+        stream_name = "Verona"
+        message_id = self.send_stream_message(
+            sender=cordelia,
+            stream_name=stream_name,
+        )
+        self.login_user(cordelia)
+
+        payload = dict(
+            message_id=message_id,
+            msg_type="whatever",
+            content='{"name": "alice", "salary": 20}',
+        )
+        self.client_post("/json/submessage", payload)
```

Comment:
We can possibly verify that this request is successful with `self.assert_json_success`